docs/docs/40-operator-guide/40-security/40-managing-credentials.md

```diff
@@ -255,58 +255,88 @@ obtain a token that is valid for 12 hours and cached for 10.
-using Workload Identity Federation.
+Kargo can be configured to authenticate to
+[Google Artifact Registry](https://cloud.google.com/artifact-registry/docs/overview)
+(GAR) repositories using
+[Workload Identity Federation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity))
+(WIF).
 
 If Kargo locates no `Secret` resources matching a repository URL, and if Kargo
-is deployed within a GKE cluster, it will attempt to use
-[Workload Identity Federation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
-to authenticate, but this relies upon some external setup. Leveraging this
-option eliminates the need to store credentials in a `Secret` resource.
-
-First, follow
-[these directions](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_on_cluster)
-to provision a new GKE cluster with Workload Identity Federation enabled or
-[these directions](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable-existing-cluster)
-to enable Workload Identity Federation on an existing GKE cluster.
-
-At this point, the `kargo-controller` `ServiceAccount` within the namespace in
-which Kargo is (or will be) installed will be associated with an _IAM principal
-identifier_, which takes the following form:
-
-```plaintext
+is deployed within a
+[Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/concepts/kubernetes-engine-overview)
+(GKE) cluster with WIF enabled, it will attempt to use it to authenticate.
+Leveraging this option eliminates the need to store credentials in a `Secret`
+resource. WIF can be enabled when creating a
+[new cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_on_cluster)
+or can be added to an
+[existing cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable-existing-cluster).
+
+:::note
+Clusters managed by
+[GKE Autopilot](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview)
+have WIF enabled automatically.
+:::
+
+With WIF enabled,
+[GCP Identity and Access Management](https://cloud.google.com/iam/docs/overview)
+(IAM) automatically understands a
+[principal](https://cloud.google.com/iam/docs/overview#principals)
+identifier of the following form to be a reference to the Kargo controller's
+_Kubernetes_ Service Account (KSA):
+
+```
 principal://iam.googleapis.com/projects/<gcp project number>/locations/global/workloadIdentityPools/<gcp project name>.svc.id.goog/subject/ns/<kargo namespace>/sa/kargo-controller
 ```
 
-Although associated with this _one_ principal, the Kargo controller acts on
+:::note
+There is no need to annotate the Kargo controller's KSA in any specific way to enable the above.
+:::
+
+#### Google Service Account Impersonation
+
+Although associated with [this _one_ principal](#kubernetes-service-account-principal), the Kargo controller acts on
```

Comment:
~~~suggestion
Because the Kargo controller acts on behalf of multiple Kargo projects, each of
which may require access to _different_ GAR repositories, when accessing a
repository on behalf of a given project, it will attempt to
[impersonate](https://cloud.google.com/iam/docs/service-account-impersonation)
a project-specific
[Google Service Account](https://cloud.google.com/iam/docs/service-accounts-create) 
(GSA). The name of the GSA that the controller will attempt to impersonate will
_always_ be of the form
`kargo-project-<kargo project name>@<gcp project name>.iam.gserviceaccount.com`.
~~~